### ISSUE TITLE
Incorrect: mismatched-arg-count with RTIC

### ISSUE THREAD

**Issue by VersBinarii** [rust-analyzer/rust-analyzer#6684]
The following example causes rust-analyzer to  incorrectly report `[rust-analyzer mismatched-arg-count] [E] Expected 1 argument, found 0` with those 2 lines:
```
let _flash = dp.FLASH.constrain();
let _rcc = dp.RCC.constrain();
```

when using latest `cortex-m-rtic = "0.6.0-alpha.0"` release.
However when providing a dummy argument into constrain() rust analyzer then correctly says that the function in fact takes no arguments.

```
#![no_std]
#![no_main]

use panic_halt as _;
use stm32f1xx_hal::stm32;

#[rtic::app(device = crate::stm32, peripherals = true)]
mod app {
    use panic_halt as _;
    use stm32f1xx_hal::prelude::*;

    #[resources]
    struct Resources {}

    #[init]
    fn init(cx: init::Context) -> init::LateResources {
        let dp = cx.device;

        let _flash = dp.FLASH.constrain();
        let _rcc = dp.RCC.constrain();

        init::LateResources {}
    }

    #[idle]
    fn idle(_cx: idle::Context) -> ! {
        loop {}
    }
}

```

**Comment by bjorn3** [comment]
Attribute proc macros are not yet supported.

**Comment by flodiebold** [comment]
As bjorn3 wrote, we don't run attribute macros yet, so `cx` and `dp` are probably of unknown type. We should keep track of the type errors in this situation and avoid issuing follow-up diagnostics.

**Comment by VersBinarii** [comment]
Thanks, i understood the issue. I'm happy to close this issue unless you wish  to use this for keeping the track of it.

**Comment by lnicola** [comment]
Proc macro attributes are tracked in https://github.com/rust-analyzer/rust-analyzer/issues/6029, please watch that and re-test when it's implemented. Closing to keep things tidy.
